src/test/java/org/javarosa/core/model/test/FormDefTest.java

```diff
@@ -497,5 +498,12 @@ public Object eval(Object[] args, EvaluationContext ec) {
 
         assertThat(deprecatedID, is(originalInstanceID));
         assertThat(newInstanceID, not(originalInstanceID));
+
+        // Subsequent initialization - nothing should change
+        editFinalizedFormScenario = Scenario.from(newFormScenario.getFormDef(), FormInitializationMode.FINALIZED_FORM_EDIT);
```

Comment:
What situations do we end initializing twice in?

Reply:
I added it just in case. If handled correctly by clients (like ODK Collect), this shouldn't happen.  
In Collect, when editing a form, `FormDef#initialize(FormInitializationMode.FINALIZED_FORM_EDIT)` should be called, and this should be the only case. The only tricky scenario I can think of is when a user edits a form, and while still in the editing process, the app is killed and restored. In that case, we would need to ensure Collect doesn't call `FormDef#initialize(FormInitializationMode.FINALIZED_FORM_EDIT)` again.  
I would need to check that to be sure, but handling it in JR ensures safety.

Reply:
I think this might be related to https://github.com/getodk/javarosa/pull/819#discussion_r2005390910 actually: depend ending on how we set `instanceId`, it might not matter if `initialize` is called multiple times.

Reply:
The thing we have to be absolutely sure of is that deprecatedId can only be set exactly once when copying over the original instance to edit.

Reply:
Ah, I see, this won't work, I don't think. We need to be able to handle edits to submissions that already have a deprecatedid. It's going to be the client's responsibility to only call FINALIZED_FORM_EDIT exactly once at copy time.

Reply:
>  We need to be able to handle edits to submissions that already have a deprecatedid.

What scenario would that be? We have this issue: https://github.com/getodk/collect/issues/6594, which states that once a form has been edited, it can't be edited again.

Reply:
That issue is about making sure that if you already have a 6th edit to an original submission, you can't make edits to that original submission, the 1st edit, the second, etc. You may only continue making edits to edit 6. Edit 6 does have a deprecatedid.

Reply:
Ahh ok so once an edit is finalized it also can be edited, right! Ok so I will unblock that scenario.